superset/dataframe.py

```diff
@@ -50,9 +52,17 @@ def size(self):
     @property
     def data(self):
         # work around for https://github.com/pandas-dev/pandas/issues/18372
-        return [dict((k, _maybe_box_datetimelike(v))
-                     for k, v in zip(self.__df.columns, np.atleast_1d(row)))
+        data = [dict((k, _maybe_box_datetimelike(v))
+                for k, v in zip(self.__df.columns, np.atleast_1d(row)))
                 for row in self.__df.values]
+        for d in data:
+            for k, v in list(d.items()):
+                # if an int is too big for Java Script to handle
+                # convert it to a string
+                if isinstance(v, int):
```

Comment:
What about moving the constant and the loop in an helper?

Reply:
Sure, will do.

Reply:
Good idea, `def handle_js_int_overflow(data)` or something